recipes/antlr4-cppruntime/all/conanfile.py

```diff
@@ -23,133 +28,126 @@ class Antlr4CppRuntimeConan(ConanFile):
         "shared": False,
         "fPIC": True,
     }
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "pkg_config"
     short_paths = True
 
-    compiler_required_cpp17 = {
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
             "Visual Studio": "16",
+            "msvc": "1923",
             "gcc": "7",
             "clang": "5",
             "apple-clang": "9.1"
-    }
-
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        # We need this wrapper to build the library on Windows properly.
```

Comment:
Why don't you set CMakeToolchain.variable["WITH_STATIC_CRT"] instead?

Reply:
You can use ``cache_variables`` if the project does not respect the CMake variables